validation/DefaultValuesValidator.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+declare(strict_types=1);
+
+namespace Google\Generator\Validation;
+
+use Google\Showcase\V1beta1\ComplianceData;
+
+require __DIR__ . '/../vendor/autoload.php';
+error_reporting(E_ALL);
+
+// Usage: php validation/DefaultValuesValidator.php
+
+$data = new ComplianceData();
+$data->setFString("fstring no presence")
+     ->setPString("P_string optional")
+     ->setPBool(False);
+$dataJson = $data->serializeToJsonString();
+assert($dataJson === ' {"fString":"fstring no presence","pString":"P_string optional","pBool":true}');
```

Comment:
1. in this assertion, `pBool` should be `false`, right?
2. Maybe do one more explicit zero-value check, that `setPString("")` returns `"pString": ""`
3. It looks like non-`optional` fields that have zero-values do not get transcoded? That should be fine (and not affect DIREGAPIC, only REGAPIC, and presumably beningly)
4. Are you planning on checking this file in as a test or otherwise? No need to, assuming we capture all this behavior in some other tests.

Reply:
No plans to check this in, this PR should be sufficient (and we can revive in the future).